cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -332,11 +333,20 @@ private static ClusterControllerClient getClusterControllerClient(DataprocConf c
   private static Compute getCompute(DataprocConf conf) throws GeneralSecurityException, IOException {
     HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
     return new Compute.Builder(httpTransport, JacksonFactory.getDefaultInstance(),
-                               new HttpCredentialsAdapter(conf.getComputeCredential()))
+                               setHttpTimeouts(new HttpCredentialsAdapter(conf.getComputeCredential()), conf))
       .setApplicationName("cdap")
       .build();
   }
 
+  private static HttpRequestInitializer setHttpTimeouts(HttpRequestInitializer requestInitializer,
+                                                        DataprocConf conf) {
+    return httpRequest -> {
+      requestInitializer.initialize(httpRequest);
```

Comment:
It seems like the `requestInitializer` will not be getting used by the Compute.Builder, is it??

Reply:
Hey, I referred [this](https://developers.google.com/api-client-library/java/google-api-java-client/errors) guide for setting timeouts.